app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java

```diff
@@ -0,0 +1,181 @@
+package com.example.bookwormadventuresdeluxe2;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.res.Resources;
+import android.widget.EditText;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.rule.ActivityTestRule;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.robotium.solo.Solo;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+
+/**
+ * Tests for login screen, cannot be
+ */
+@RunWith(AndroidJUnit4.class)
+public class LoginActivityTest
+{
+    private Solo solo;
+
+    private Context appContext;
+    private Resources r;
+
+    private int shortWait = 1000;
+    private int longWait = 4000;
+
+    private EditText emailText;
+    private EditText passwordText;
+
+    @Rule
+    public ActivityTestRule<LoginActivity> rule =
+            new ActivityTestRule<>(LoginActivity.class, true, true);
+
+    /**
+     * Runs before all tests and creates solo instance.
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception{
+        solo = new Solo(getInstrumentation(),rule.getActivity());
+
+        /* Gets context for app resources */
+        appContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+
+        /* Gets resource files */
+        r = appContext.getResources();
+
+        emailText = (EditText) solo.getView(R.id.login_email);
+        passwordText = (EditText) solo.getView(R.id.login_password);
+    }
+
+    /**
+     * Gets the activity
+     *
+     * @throws Exception
+     */
+    @Test
+    public void start() throws Exception {
+        Activity activity = rule.getActivity();
+    }
+
+    /**
+     * Tests going to CreateAccountActivity
+     */
+    @Test
+    public void createAccountButtonTest()
+    {
+        solo.clickOnButton(r.getString(R.string.create_account));
+        solo.assertCurrentActivity(r.getString(R.string.wrong_activity), CreateAccountActivity.class);
+
+        solo.sleep(shortWait);
+    }
+
+    /**
+     * Tests login button functionality with no input
+     */
+    @Test
+    public void emptyLoginTest()
```

Comment:
Can we assert that the error message is shown and the red ! icon? This is an important part of this test I think

Reply:
sec i am doing so